float32 bound
int32 size
float32 factor
float32[] path
